### PR TITLE
CMR-3365 Convert VerticalSpatialDomainType to an enum

### PR DESCRIPTION
**Pull request by mschmele**
There is still some cleanup that needs to be done (mostly just moving stuff into an iso-shared/spatial.clj file), as well as a rebase but I wanted the chance to get eyes on this

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+   "MaximumDepth"
+   "MinimumAltitude"
+   "MinimumDepth"])
+
+(defn vertical-spatial-domain-type-is-valid?
+  "Return true or false based on whether or not the given value matches
+   one of the valid values in valid-vertical-spatial-domain-types."
+  [vs-domain-type]
+  (let [squished-vs-domain-type (-> vs-domain-type
+                                    (string/split #"( |_)")
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj]: Lines 35 and 36 can be replaced with `(string/replace #"( |_)" "")`

File: umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10/spatial.clj [hunk 2]
```diff
@@ -57,6 +59,14 @@
      [:Minimum_Value (:MinimumValue coord)]
      [:Maximum_Value (:MaximumValue coord)]]))
 
+(defn- convert-vertical-spatial-domains
+  "Validate and convert vertical spatial domains to UMM-C v1.10.0"
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10/spatial.clj]: This docstring is confusing. You're converting the vertical spatial domains to dif10 format with '_'

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10/spatial.clj [hunk 3]
```diff
@@ -53,8 +57,9 @@
      :HorizontalSpatialDomain      (let [[geom] (select spatial "Geometry")]
                                      {:Geometry (parse-geometry geom)
                                       :ZoneIdentifier (value-of spatial "Zone_Identifier")})
-     :VerticalSpatialDomains       (for [vert-elem (select spatial "Vertical_Spatial_Info")]
-                                     (fields-from vert-elem :Type :Value))
+     :VerticalSpatialDomains       (->> (select spatial "Vertical_Spatial_Info")
+                                        spatial-conversion/convert-vertical-spatial-domains-from-xml
+                                        (map #(update % :Type (fn [x] (string/replace x "_" " ")))))
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10/spatial.clj]: I think what's being done in the map should be done in `spatial-conversion/convert-vertical-spatial-domains-from-xml`. That function should squish them if needed.

File: umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj [hunk 4]
```diff
@@ -1,12 +1,58 @@
...
+(defn drop-invalid-vertical-spatial-domains
+  "Any VerticalSpatialDomain with a Type not in valid-vertical-spatial-domain-types
+   is not considered valid according to UMM spec v1.10.0 and should be dropped.
+   This behavior will eventually generate errors"
+  [vertical-spatial-domains]
+  (filter
+   #(vertical-spatial-domain-type-is-valid? (:Type %))
+   vertical-spatial-domains))
+
+(defn convert-vertical-spatial-domains-from-xml
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj]: I think what you should do in this function is get all the types and convert them to something valid if possible - i.e. replace the _ with a space or add a space where needed. And also drop all of the invalid values.

File: umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj (hunk 1 above)
Comment (laurenfrederick) [reply umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj]: But also, if MaximumAltitude is the original value, it'll say it's valid when it's not really valid, right?

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj [hunk 5]
```diff
@@ -1186,7 +1186,20 @@
   (is (= "COMPLETE"
          (:CollectionProgress
            (vm/migrate-umm {} :collection "1.9" "1.10"
-                          {:CollectionProgress "COMPLETE"}))))))
+                          {:CollectionProgress "COMPLETE"}))))
+  (testing "VerticalSpatialDomains migration from 1.9.0 to 1.10.0"
+    (let [vsds {:VerticalSpatialDomains [{:Type "An Invalid Type"
```
Comment (laurenfrederick) [inline umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj]: What happens if we have one with a type "AtmosphereLayer"?

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10/spatial.clj [hunk 6]
```diff
@@ -52,14 +52,14 @@
   "Returns UMM-C spatial map from DIF 10 XML document."
   [doc]
   (let [[spatial] (select doc "/DIF/Spatial_Coverage")]
+    (proto-repl.saved-values/save 7)
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10/spatial.clj]: Remove this.

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj (hunk 5 above)
Comment (mschmele) [reply umm-spec-lib/test/cmr/umm_spec/test/migration/version_migration.clj]: Added a test and it passes

File: umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj [hunk 7]
```diff
@@ -51,7 +39,8 @@
   [vertical-spatial-domains]
   (->> vertical-spatial-domains
        (map #(xml-parse/fields-from % :Type :Value))
-       drop-invalid-vertical-spatial-domains))
+       drop-invalid-vertical-spatial-domains
+       (map #(update % :Type (fn [x] (string/replace x "_" " "))))))
```
Comment (laurenfrederick) [inline umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj]: So I think you could do this line before `drop-invalid-vertical-spatial-domains` and then you wouldn't have to do the `(string/replace...)` on line 24.

File: umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj (hunk 4 above)
Comment (mschmele) [reply umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj]: Yup, I added that `map` from the Dif10 conversion. Fixed.

File: umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj [hunk 8]
```diff
@@ -110,12 +110,12 @@
...
   "Returns the expected ECHO10 SpatialExtent for comparison with the umm model."
   [spatial-extent]
-  (let [spatial-extent (conversion-util/prune-empty-maps spatial-extent)]
-    (if (get-in spatial-extent [:HorizontalSpatialDomain :Geometry])
-      (update-in spatial-extent
-                 [:HorizontalSpatialDomain :Geometry]
-                 conversion-util/geometry-with-coordinate-system)
-      spatial-extent)))
+ (as-> spatial-extent se
+     (conversion-util/prune-empty-maps se)
```
Comment (daniel-zamora) [inline umm-spec-lib/src/cmr/umm_spec/test/echo10_expected_conversion.clj]: Fix indentation

File: umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj [hunk 9]
```diff
@@ -294,6 +294,7 @@
    be compared to the actual translation."
   [umm-coll]
   (-> umm-coll
+<<<<<<< HEAD
```
Comment (daniel-zamora) [inline umm-spec-lib/src/cmr/umm_spec/test/iso_smap_expected_conversion.clj]: fix

File: umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_smap/spatial.clj [hunk 10]
```diff
@@ -2,6 +2,8 @@
   "Functions for parsing UMM spatial records out of ISO SMAP XML documents."
   (:require
    [cmr.common.xml.parse :refer :all]
+   [cmr.umm-spec.spatial-conversion :as spatial-conversion]
+   [cmr.umm-spec.xml-to-umm-mappings.iso19115-2.spatial :as iso-19115-2-spatial]
    [cmr.common.xml.simple-xpath :refer [select]]
```
Comment (daniel-zamora) [inline umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_smap/spatial.clj]: sort

File: umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj [hunk 11]
```diff
@@ -1,8 +1,10 @@
 (ns cmr.umm-spec.spatial-conversion
   "Defines functions that convert umm spec spatial types to spatial lib spatial shapes."
   (:require
+   [clojure.string :as string]
    [cmr.spatial.line-string :as ls]
    [cmr.spatial.mbr :as mbr]
+   [cmr.common.xml.parse :as xml-parse]
```
Comment (daniel-zamora) [inline umm-spec-lib/src/cmr/umm_spec/spatial_conversion.clj]: sort

**Comment by mschmele** [comment]
@sxu123 That all is up to date already

**Pull request merged**